### PR TITLE
Add Developer API to UCSBSubjects

### LINKED ISSUE #10: Implement UCSBSubjectsService
Implement UCSBSubjectsService

There is a stub in the code for a UCSBSubjectsService. Currently it 
delivers a `List<UCSBSubject>` of three hard coded `UCSBSubject` objects.

Modify it to, instead, use the UCSB Developer API to get 
subjects from the endpoint

* `https://api.ucsb.edu/students/lookups/v1/subjects?includeInactive=false`

You can use the documentation at <https://developer.ucsb.edu>.

If you need credentials; Prof. Conrad can supply these to your team (ask on Slack.)

You can also apply for an account directly: ask only for these APIs (which are all auto-approved).  Asking for more than this could delay your request.

* Academics: Academic Curriculums 
* Academics: Academic Quarter Calendar
* Students: Student Record Code Lookups

Use the code from the team01 assignment as a model.

# Acceptance Criteria

- [x] Correctly calls UCSB Developer API to get the subjects and returns a response in the right format
- [x] 100% test coverage
- [x] 100% mutation coverage

### PR DESCRIPTION
**Pull request by LucienLuc**
# Overview

UCSB Subjects /get endpoint calls UCSB developer API. Added tests. 100% coverage and mutation.

# Issues Addressed

Closes #10

### CODE REVIEW

**Review by lukeyoffe — APPROVED** [review]
LGTM!

File: src/main/java/edu/ucsb/cs156/example/services/UCSBSubjectsService.java [hunk 1]
```diff
@@ -1,57 +1,63 @@
...
+    public UCSBSubjectsService(RestTemplateBuilder restTemplateBuilder) {
+        restTemplate = restTemplateBuilder.build();
+    }
+    public List<UCSBSubject> get() throws JsonProcessingException {
 
         // TODO: Replace with a service that will call the UCSB
         // developer API at the endpoint
         // https://api.ucsb.edu/students/lookups/v1/subjects?includeInactive=false
         // convert the JSON to a List<UCSBSubjects> object
         // return that object.
```
Comment (asharma381) [inline src/main/java/edu/ucsb/cs156/example/services/UCSBSubjectsService.java:47]: Is this still a TODO or has this been addressed already? If so, we can delete this.

**Review by asharma381 — APPROVED** [review]
Approved! Just a minor comment about line 43, do we still need that TODO comment. It looks like it's already been addressed by the code below. Other than that looks good to me!

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -28,3 +28,5 @@ spring.mvc.format.date-time=iso
 
 app.startQtrYYYYQ=${START_QTR:${env.START_QTR:20221}}
 app.endQtrYYYYQ=${END_QTR:${env.END_QTR:20222}}
+
+app.ucsb.api.consumer_key=${UCSB_API_KEY:${env.UCSB_API_KEY:see-instructions-in-readme}}
```
Comment (NickWang11235) [inline src/main/resources/application.properties:32]: Is is supposed to be `see-instructions-in-readme` here?

**Review by NickWang11235 — DISMISSED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (LucienLuc) [reply src/main/resources/application.properties:32]: Yes

**Review by asharma381 — APPROVED** [review]
Great, looks good to me! Addressed the comment earlier.

**Review by sidd328 — APPROVED** [review]
LGTM (Just check for unnecessary imports in your editor, but if none then its good to go)

**Review by pconrad — APPROVED** [review]
LGTM!  Merging for 10 points

**Pull request merged**